Revit_Core_Engine/Create/Solid.cs

```diff
@@ -94,5 +94,62 @@ public static Solid Solid(IList<CurveLoop> boundaries, double bottomElevation, d
         }
 
         /***************************************************/
+
+        [Description("Create Solid object from ViewPlan (based on cropbox and view range).")]
+        [Input("view", "ViewPlan to get the solid from.")]
+        [Input("transform", "Transform of the link model.")]
+        [Output("solid", "Solid object created from the ViewPlan.")]
+        public static Solid Solid(this ViewPlan view, Transform transform)
+        {
+            if (view == null || transform == null)
+                return null;
+
+            //bbox of current view
+            Document doc = view.Document;
+            PlanViewRange viewRange = view.GetViewRange();
+            BoundingBoxXYZ viewBBox = view.CropBox;
+            if (!view.CropBoxActive)
+            {
+                viewBBox.Min = new XYZ(-1e6, -1e6, -1e6);
+                viewBBox.Max = new XYZ(1e6, 1e6, 1e6);
+            }
+
+            //topElevation (from the CutPlane)
+            Level topLevel = doc.GetElement(viewRange.GetLevelId(PlanViewPlane.CutPlane)) as Level;
+            double topElevation;
+            if (topLevel == null)
+            {
+                topElevation = 1e6;
```

Comment:
Sanity check: isn't that beyond Revit's acceptable range?

Reply:
I checked that and bounding boxes is created for that range